EXILED/Exiled.API/Features/Respawn.cs

```diff
@@ -231,49 +267,79 @@ public static void SummonNtfChopper()
         /// <remarks>This will also trigger Music effect.</remarks>
         public static void SummonChaosInsurgencyVan()
         {
-            if (TryGetWaveBase(Faction.FoundationEnemy, out SpawnableWaveBase wave))
+            if (TryGetWaveBase(SpawnableFaction.ChaosWave, out SpawnableWaveBase wave))
                 PlayEffect(wave);
         }
 
         /// <summary>
         /// Grants tickets to a <see cref="SpawnableTeamType"/>.
         /// </summary>
-        /// <param name="team">The <see cref="SpawnableTeamType"/> to grant tickets to.</param>
+        /// <param name="faction">The <see cref="SpawnableTeamType"/> to grant tickets to.</param>
         /// <param name="amount">The amount of tickets to grant.</param>
-        public static void GrantTickets(Faction team, int amount)
+        /// <returns>Success.</returns>
+        public static bool GrantTokens(Faction faction, int amount)
         {
-            if (TryGetWaveBase(team, out SpawnableWaveBase wave) && wave is ILimitedWave limitedWave)
-                limitedWave.RespawnTokens += amount;
+            if (TryGetWaveBases(faction, out IEnumerable<SpawnableWaveBase> waveBases))
+            {
+                foreach (ILimitedWave limitedWave in waveBases.Select(x => (ILimitedWave)x))
```

Comment:
```suggestion
                foreach (ILimitedWave limitedWave in waveBases.OfType<ILimitedWave>())
```
This should work. Yeah, currently we don't have waves that don't inherit this interface but... idk why nw has added it. I thimk there was some point